internal/testplanet/uplink_test.go

```diff
@@ -37,3 +40,72 @@ func TestUploadDownload(t *testing.T) {
 
 	assert.Equal(t, expectedData, data)
 }
+
+func TestDownloadWithSomeNodesOffline(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 5, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		// first, upload some remote data
+		ul := planet.Uplinks[0]
+		satellite := planet.Satellites[0]
+
+		testData := make([]byte, 1*memory.MiB)
+		_, err := rand.Read(testData)
+		require.NoError(t, err)
+
+		err = ul.UploadWithConfig(ctx, satellite, &uplink.RSConfig{
+			MinThreshold:     2,
+			RepairThreshold:  3,
+			SuccessThreshold: 4,
+			MaxThreshold:     5,
+		}, "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		// get a remote segment from pointerdb
+		pdb := satellite.Metainfo.Service
+		listResponse, _, err := pdb.List("", "", "", true, 0, 0)
+		require.NoError(t, err)
+
+		var path string
+		var pointer *pb.Pointer
+		for _, v := range listResponse {
+			path = v.GetPath()
+			pointer, err = pdb.Get(path)
+			require.NoError(t, err)
+			if pointer.GetType() == pb.Pointer_REMOTE {
```

Comment:
What does this do?

Reply:
This makes sure we're getting a pointer to a remote segment (vs. a pointer to an inline segment, which the pdb.Get call could also potentially return). We want a remote segment because that means its pieces will be on storage nodes, and we want to test the download after taking some of these nodes offline.

Reply:
but you're only going to break out of `for _, v := range listResponse {`

Reply:
right, so at that point we'll have `pointer` assigned to a remote pointer

Reply:
ah. somehow my brain could process all that because it fails if all pointers are inline.  not a bid deal.  LGTM.